Add `max_tag_value_length` support to Jaeger exporter

owais opened the pull request:
# Description

This commit adds ability to specifiy maximum length for string
attributes to the Jaeger exporter. This is similar to how Zipkin
exporter supports `max_tag_value_length`.


## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new tests

# Does This PR Require a Contrib Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

codeboten reviewed (commented):
The changes to add `max_attr_value_length` look good, what happened w/ `test_jaeger_exporter_protobuf.py`?

lzchen commented on CHANGELOG.md:
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
+### Added
+- Add `max_attr_value_length` support to Jaeger exporter
+  ([#1633])(https://github.com/open-telemetry/opentelemetry-python/pull/11633)
```
```suggestion
  ([#1633])(https://github.com/open-telemetry/opentelemetry-python/pull/1633)
```

owais commented:
Oh! My bad. I had added another test file and later deleted it but looks like I deleted the wrong file and in the process lost the new tests I wrote. I'll go back and write tests for it again now 🤦

codeboten reviewed (approved):
@owais thanks for fixing!
